web/3-buttonbadge/piratebadge.dart

```diff
@@ -6,22 +6,34 @@ import 'dart:html';
 
 ButtonElement genButton;
 
-void  main() {
-  querySelector('#inputName').onInput.listen(updateBadge);
+main() {
+  var inputField = querySelector('#inputName');
   genButton = querySelector('#generateButton');
-  genButton.onClick.listen(generateBadge);
+  initEventListeners(inputField);
```

Comment:
Also, does this even work? My understanding is that the second await for () won't even run until the first await for is finished, which will never happen.